instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/AlternativeRouteSelectionTest.kt

```diff
@@ -42,124 +33,105 @@ class AlternativeRouteSelectionTest : BaseTest<BasicNavigationViewActivity>(
-
-    protected lateinit var mapboxNavigation: MapboxNavigation
-
-    protected lateinit var routeLineApi: MapboxRouteLineApi
-    protected lateinit var routeLineView: MapboxRouteLineView
-
-    private val myResourceIdler =
-        CountingIdlingResource("AlternativeRouteSelectionTestResource")
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var routeLineApi: MapboxRouteLineApi
+    private lateinit var routeLineView: MapboxRouteLineView
 
     @Before
     fun setUp() {
-        initIdlingResource = MapStyleInitIdlingResource(activity.binding.mapView)
-        initIdlingResource.register()
-        IdlingRegistry.getInstance().register(myResourceIdler)
-        Espresso.onIdle()
-
-        directionsResponse = getRoute(activity)
-        val coordinates = directionsResponse.waypoints()?.mapNotNull { it.location() } ?: listOf()
-        val route = directionsResponse.routes()[0]
-
         mapboxNavigation = MapboxNavigation(
             NavigationOptions.Builder(activity)
                 .accessToken(getMapboxAccessTokenFromResources(activity))
                 .build()
         )
-
-        routeProgressIdlingResource = RouteProgressStateIdlingResource(
-            mapboxNavigation,
-            RouteProgressState.TRACKING
-        )
-
-        runOnMainSync {
-            mockLocationUpdatesRule.pushLocationUpdate {
-                latitude = coordinates.first().latitude()
-                longitude = coordinates.first().longitude()
-            }
-            mockLocationReplayerRule.playRoute(route)
-
-            mapboxNavigation.setRoutes(directionsResponse.routes())
-        }
     }
 
     @After
     fun tearDown() {
-        initIdlingResource.unregister()
+        mapboxNavigation.onDestroy()
     }
 
     @Test
-    fun selectAlternateRoute() {
-        addRouteLine()
-        routeProgressIdlingResource.register()
+    fun expect_route_apis_to_be_updated_with_alternatives() {
+        // Initialize with a set of routes with alternatives. Later we will
+        // update the routes with a new primary route and verify it is all set.
+        setupRouteWithAlternatives()
+        verifyRouteLineIsUpdatedWithAlternatives()
+
+        // Wait for route progress to come back with an alternative set.
+        val routeProgressCount = CountDownLatch(1)
 
         runOnMainSync {
-            myResourceIdler.increment()
-            assertEquals(3, routeLineApi.getRoutes().size)
-
-            val soonToBePrimaryRoute = routeLineApi.getRoutes()[1]
-            val updatedRoutes = routeLineApi.getRoutes()
-                .filter { it != soonToBePrimaryRoute }
-                .toMutableList()
-                .also {
-                    it.add(0, soonToBePrimaryRoute)
-                }
+            // Create routes where an alternative is set to the primary route position.
+            val updatedRoutes = routeLineApi.getRoutes().toMutableList()
```

Comment:
going to wait for a circle-ci run to pass. but it would be better if used `mapboxNavigation.getRoutes()` here. That would make the core sdk the single source of truth.

Reply:
The point of the test is to make sure the route line class and core agree. The calls should mimic what would happen when a user is selecting an alternative route. I suggest leaving it as is.

Reply:
My thinking here, is that `instrumentation_tests` should represent recommended implementations.

But you're right, it seems like a valid use case to take the routes from the RouteLineApi, and pass them back into MapboxNavigation and expect everything to become in sync. Updated again.

This will give us an extra run, the first one could have been green by coincidence. But if we have 2 in a row that are green, we probably fixed the flaky test